src/wp-includes/post.php

```diff
@@ -69,8 +69,17 @@ function create_initial_post_types() {
 		)
 	);
 
-	// Enhance page editor for block themes by rendering template and content blocks.
-	if ( wp_is_block_theme() && current_theme_supports( 'block-templates' ) ) {
+	/*
+	 * Enhance page editor for block themes by rendering template and content blocks.
+	 *
+	 * Theme support for block templates is added on the 'after_setup_theme' hook in
+	 * wp_enable_block_templates(). So post type support will only be added on the second
+	 * pass of this function on the 'init' hook.
+
+	 * Note: The order of these checks is important, a doing_it_wrong notice will be fired
+	 * if the block theme check is done first.
+	 */
+	if ( current_theme_supports( 'block-templates' ) && wp_is_block_theme() ) {
```

Comment:
This order avoids the `_doing_it_wrong()` only because `block-templates` support is added on `after_theme_setup` so `wp_is_block_theme()` cannot be called too early. While this works, it still seems a little fragile to me and has me wondering if moving this `add_post_type_supports()` call to a standalone function that is called on `after_setup_theme` would be more reasonable. It also would make it easier for people who want to opt-out of this default to unhook the callback (for better or worse).